sycl/test/matrix/matrix-tf32-test.cpp

```diff
@@ -0,0 +1,164 @@
+          size_t NUM_ROWS_B, size_t NUM_COLS_B, size_t NUM_ROWS_C,
+          size_t NUM_COLS_C>
+void matrix_multiply(big_matrix<T1, NUM_ROWS_C, NUM_COLS_C> &C,
+                     big_matrix<T2, NUM_ROWS_A, NUM_COLS_A> &A,
+                     big_matrix<T2, NUM_ROWS_B, NUM_COLS_B> &B) {
+  size_t M = NUM_ROWS_C;
+  size_t N = NUM_COLS_C;
+  size_t K = NUM_COLS_A;
+
+  assert(NUM_ROWS_C == NUM_ROWS_A && NUM_COLS_A == NUM_ROWS_B);
+  size_t NDRangeM = M / TM;
+  size_t NDRangeN = N / TN;
+  buffer<float, 2> bufA(A.get_data(), range<2>(M, K));
+  buffer<float, 2> bufB(B.get_data(), range<2>(K, N));
+  buffer<float, 2> bufC((float *)C.get_data(), range<2>(M, N));
+
+  queue q;
+  q.submit([&](handler &cgh) {
+     auto accC = bufC.get_access<access::mode::read_write>(cgh);
+     auto accA = bufA.get_access<access::mode::read_write>(cgh);
+     auto accB = bufB.get_access<access::mode::read_write>(cgh);
+
+     cgh.parallel_for<class imatrix>(
+         nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}),
+         [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]]
+
+         {
+           // The matrix API has to be accessed by all the workitems in a
+           // subgroup these functions will be called once by the subgroup no
+           // code divergence between the workitems
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, precision::tf32, use::a, TM, TK,
+                        layout::row_major>
+               sub_a;
+           joint_matrix<sub_group, precision::tf32, use::b, TK, TN,
+                        layout::row_major>
+               sub_b;
+           joint_matrix<sub_group, float, use::accumulator, TM, TN> sub_c;
+           joint_matrix_load(sg, sub_c,
+                             accC.get_pointer() + (sg_startx * TM) * N +
+                                 sg_starty / SG_SZ * TN,
+                             N, layout::row_major);
+           joint_matrix_fill(sg, sub_a, 42);
+           for (int k = 0; k < K; k += TK) {
+             joint_matrix_load(
+                 sg, sub_a, accA.get_pointer() + (sg_startx * TM) * K + k, K);
+             joint_matrix_load(
+                 sg, sub_b,
+                 accB.get_pointer() + (k) * (N) + sg_starty / SG_SZ * TN, N);
+             // If no rounding to tf32 function is called, joint_matrix_mad
+             // function will work on truncated floats.
+             auto wi_data_a =
+                 sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_a);
+             for (int i = 0; i < wi_data_a.length(); i++) {
+               wi_data_a[i] = round_to_tf32(wi_data_a[i]);
```

Comment:
nit, but shouldn't this type of usage of `round_to_tf32` really use `joint_matrix_apply`?

Reply:
Yes, I will use joint_matrix_apply for A and leave the loop way for B to show both usages.